Gems/Multiplayer/Code/Source/Components/SimplePlayerSpawnerComponent.cpp

```diff
@@ -0,0 +1,124 @@
+                ->Version(1)
+                ->Field("PlayerSpawnable", &SimplePlayerSpawnerComponent::m_playerSpawnable)
+                ->Field("SpawnPoints", &SimplePlayerSpawnerComponent::m_spawnPoints)
+                ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<SimplePlayerSpawnerComponent>(
+                               "Simple Network Player Spawner",
+                               "A simple player spawner that comes included with the Multiplayer gem. Attach this component to any level's root entity which needs to spawn a network player."
+                                        "If no spawn points are provided the network players will be spawned at the world-space origin.")
+                           ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                           ->Attribute(AZ::Edit::Attributes::Category, "Multiplayer")
+                           ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/NetBinding.svg")
+                           ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Level"))
+                           ->DataElement(
+                               AZ::Edit::UIHandlers::Default,
+                               &SimplePlayerSpawnerComponent::m_playerSpawnable,
+                               "Player Spawnable Asset",
+                               "The network player spawnable asset which will be spawned for each player that joins.")
+                           ->DataElement(
+                               AZ::Edit::UIHandlers::Default,
+                               &SimplePlayerSpawnerComponent::m_spawnPoints,
+                               "Spawn Points",
+                               "Networked players will spawn at the spawn point locations in order. If there are more players than spawn points, the new players will round-robin back starting with the first spawn point.")
+                ;
+            }
+        }
+    }
+
+    void SimplePlayerSpawnerComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("MultiplayerSpawnerService"));
+    }
+
+    void SimplePlayerSpawnerComponent::GetIncompatibleServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("MultiplayerSpawnerService"));
+    }
+
+    NetworkEntityHandle SimplePlayerSpawnerComponent::OnPlayerJoin(
+        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const MultiplayerAgentDatum& agentDatum)
+    {
+        const PrefabEntityId prefabEntityId(AZ::Name(m_playerSpawnable.m_spawnableAsset.GetHint().c_str()));
+        AZ::Transform transform = AZ::Transform::Identity();
+
+        if (m_spawnPoints.empty())
+        {
+            AZLOG_WARN("SimplePlayerSpawnerComponent is missing spawn points. Spawning new player at the origin.")
```

Comment:
Perhaps, we can use this entity position in the level as the fallback location if nothing is specified?

Reply:
Or spawn at the origin by default as the fallback?

Reply:
Spawn at origin is good. I wouldn't get too fancy, but if we had a navmesh or a did a physx scene query you could also make sure you don't spawn a player underground. Honestly I'm inclined to leave this as-is, just spawn at 0,0,0, since this should only ever be used as a quick getting started crutch.

Reply:
Yeah, I think spawning at the origin is the safest bet. Done deal!